2_build_systems/2_build_systems.md

```diff
@@ -0,0 +1,240 @@
+## Compilers
+
+### What is a compiler?
+
+A compiler is a special program that translates a programming language's source code into machine code, bytecode or another programming language. The source code is typically written in a high-level, human-readable language such as C/C++.
+
+### Compiler Directives 
+
+Compiler directives refer to statement written in the source code of a program that lets the programmer instruct the compiler to perform a specific operation within the compilation phase itself.
+
+For example, the ```#include``` directive incorporates the contents of a header file to the compilation. Similarly, The ```#define``` directive is used to define a macro.
+
+### Different types of files during compilation
+
+<p align="center">
+  <img src="../../assets/images/Compilation_Stages.png" width="800" />
+</p>
+
+The compilation process transforms a human-readable code into a machine-readable format. For a programming language, it happens before a program starts executing to check the syntax and semantics of the code. The compilation process involves four steps: pre-processing, compiling, assembling, and linking then, we run the obtained executable file to get an output on the screen.
```

Comment:
Can add the detailed explanation for each compilation process steps.

Reply:
Resolved in latest commit